velox/docs/functions/spark/string.rst

```diff
@@ -112,6 +112,31 @@ String Functions
         SELECT levenshtein('kitten', 'sitting', 10); -- 3
         SELECT levenshtein('kitten', 'sitting', 2); -- -1
 
+.. spark:function:: locate(substring, string, start) -> integer
+
+    Returns the position of the first occurrence of ``substring`` in given ``string``
+    after position ``start``. The search is from the beginning of ``string`` to the end.
+    The given ``start`` and return value are 1-based. The following rules are listed in order of priority:
+
+    Returns 0 if ``start`` is NULL. Returns NULL if ``substring`` or ``string`` is NULL.
```

Comment:
Do these rules match Spark? Any particular version or all versions are having the same behavior?

Reply:
Yes, it matches Spark, and we are testing against Spark-3.5.1. The versions before Spark-3.5.1 have the same behavior.

https://github.com/apache/spark/blame/v3.5.1/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala#L1434-L1456